suite-native/module-device-settings/src/hooks/useSettingsProtection.ts

```diff
@@ -0,0 +1,45 @@
+import { useCallback, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+
+import { useNavigation } from '@react-navigation/native';
+
+import { selectIsDeviceConnected, selectIsDeviceDiscoveryActive } from '@suite-common/wallet-core';
+import {
+    AuthorizeDeviceStackRoutes,
+    DeviceStackRoutes,
+    RootStackParamList,
+    RootStackRoutes,
+    SettingsStackParamList,
+    StackToStackCompositeNavigationProps,
+} from '@suite-native/navigation';
+
+type NavigationProps = StackToStackCompositeNavigationProps<
+    SettingsStackParamList,
+    DeviceStackRoutes.DeviceSettings,
+    RootStackParamList
+>;
+
+export const useSettingsProtection = (isProtectionEnabled = false) => {
+    const navigation = useNavigation<NavigationProps>();
+
+    const isDiscoveryRunning = useSelector(selectIsDeviceDiscoveryActive);
+    const isDeviceConnected = useSelector(selectIsDeviceConnected);
+
+    const handleConnectDevice = useCallback(() => {
+        navigation.navigate(RootStackRoutes.AuthorizeDeviceStack, {
+            screen: AuthorizeDeviceStackRoutes.ConnectAndUnlockDevice,
+            params: {
+                onCancelNavigationTarget: {
+                    name: RootStackRoutes.DeviceSettingsStack,
+                    params: { screen: DeviceStackRoutes.DeviceSettings },
+                },
+            },
+        });
+    }, [navigation]);
+
+    useEffect(() => {
+        if (!isDeviceConnected && isProtectionEnabled) return handleConnectDevice();
+    }, [handleConnectDevice, isDeviceConnected, isProtectionEnabled]);
+
+    return { isDiscoveryRunning, isDeviceConnected, handleConnectDevice };
```

Comment:
Remove `handleConnectDevice` it is not used anywhere outside of this hook.
```suggestion
    return { isDiscoveryRunning, isDeviceConnected };
```

Reply:
https://github.com/trezor/trezor-suite/pull/17545/commits/6bd8e85bd5b268b1f41c88f5ba67b95464056be0